gpcontrib/arenadata_toolkit/src/dbsize.c

```diff
@@ -366,7 +368,33 @@ Datum adb_get_relfilenodes(PG_FUNCTION_ARGS)
 }
 
 PG_FUNCTION_INFO_V1(adb_hba_file_rules);
-Datum adb_hba_file_rules(PG_FUNCTION_ARGS)
+Datum
+adb_hba_file_rules(PG_FUNCTION_ARGS)
 {
 	return pg_hba_file_rules(fcinfo);
 }
+
+/*	*/
+int64
+dbsize_calc_size(Oid relid)
+{
+	Relation	rel;
+	int64		size = 0;
+
+	rel = try_relation_open(relid, AccessShareLock, false);
```

Comment:
Locking a table (which is considered altered by the filters of `arenadata_toolkit.tracking_get_track()`) in ACCESS EXCLUSIVE lock will make `arenadata_toolkit.tracking_get_track()` hanging till the lock is removed. Though we do not select anything from the relation, only doing `stat` for its files.
Are we Ok with it?